src/main/java/frc/robot/subsystems/drivetrain/SwerveDrive.java

```diff
@@ -0,0 +1,330 @@
+                SmartDashboard.putString("EncoderSlippageResult [WHEEL " + i + "]", "WHEEL: " + i + "is BAD! Error is: " + error);
+            }
+        }
+    }
+
+
+    public void printEncoderReadings() {
+        System.out.println("THE VALUES ARE: " + getModule(0).getSelectedSensorPosition() + ", " + getModule(1).getSelectedSensorPosition() + ", " + getModule(2).getSelectedSensorPosition() + ", " + getModule(3).getSelectedSensorPosition());
+    }
+
+    public void printJustInCase() {
+        System.out.println("-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*");
+        System.out.println(" [NOTICE ME!!] TAKE A PICTURE OF THIS WITH YOUR PHONE TO FIX THE SWERVE [NOTICE ME!!]");
+        System.out.println("-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*");
+        System.out.println("To fix the wheels rotation, reset zero positions by following these steps, 1. open FRC DRIVER STATION app, disable the robot, 2. rotate the wheel so that the black part of the wheel is pointing towards the direction of the drawn y arrow (all wheels are pointing to the same direction), 3. Take another picture of this message again because you have changed the values by turning the wheels, On the left side of the coding software, under the robot-2022 folder -> src -> -> main -> java -< frc.robot -> double click the Constants class (painted in blue) and find the line ZERO_POSITIONS =" + Arrays.toString(Constants.SwerveModule.ZERO_POSITIONS) + "  -> and Change the values inside the brackets to the values:");
+        printEncoderReadings();
+    }
+
+    /**
+     * Set all the modules to their desired positions without much velocity for smoothing purposes.
+     *
+     * @param vx  velocity for the x-axis. [m/s]
+     * @param vy  velocity for the x-axis. [m/s]
+     * @param rot rotational velocity counter-clockwise positive. [rad/s]
+     */
+    public void noSpeedHolonomicDrive(double vx, double vy, double rot) {
+        ChassisSpeeds speeds = fieldOriented ?
+                ChassisSpeeds.fromFieldRelativeSpeeds(vx, vy, rot, Robot.getAngle()) :
+                new ChassisSpeeds(vx, vy, rot);
+        noSpeedSetStates(kinematics.toSwerveModuleStates(speeds));
+    }
+
+    /**
+     * Set all the modules to their desired positions without much velocity for smoothing purposes.
+     *
+     * @param states desired angles and velocities for each module.
+     */
+    public void noSpeedSetStates(SwerveModuleState[] states) {
+        for (int i = 0; i < 4; i++) {
+            double diff = Math.abs(states[i].angle.minus(getModule(i).getAngle()).getRadians());
+            states[i] = SwerveModuleState.optimize(states[i], getModule(i).getAngle());
+            getModule(i).setAngle(states[i].angle);
+            getModule(i).setVelocity(states[i].speedMetersPerSecond * Math.abs(Math.cos(diff)));//comment this out
+        }
+    }
```

Comment:
You can incorporate this part of the code in `holonomicDrive` method (it's just an if statement)